CSC8503CoreClasses/GameObject.h

```diff
@@ -22,7 +22,7 @@ namespace NCL::CSC8503 {
 
 	class GameObject	{
 	public:
-		GameObject(CollisionLayer = NoSpecialFeatures, const std::string& name = "");
+		GameObject(const std::string& name = "", CollisionLayer = NoSpecialFeatures);
```

Comment:
Gonna be a pain to re-write my stuff this way round, not sure why we're changing it back round but ok sure

Reply:
I think having it as collision layer first, name second makes more sense, because of how default parameters work: it's more likely that a gameobject would not be given a name, than it is to not be given a collision layer. 

In this way round, every game object, if it's going to be given a collision layer, must also be given a name (even if it's a restatement of "" - which would defeat the point of the default parameter). 

I changed it to collisionLayer first to make it easier to instantiate gameObjects.